pose_est_nets/utils/plotting_utils.py

```diff
@@ -351,94 +257,153 @@ def predict_videos(
+            )
+
+    # save csv file of predictions in DeepLabCut format
+    num_joints = model.num_keypoints
+    predictions = np.zeros((keypoints_np.shape[0], num_joints * 3))
+    predictions[:, 0] = np.arange(keypoints_np.shape[0])
+    # put x vals back in original pixel space
+    x_resize = cfg.data.image_resize_dims.width
+    x_og = cfg.data.image_orig_dims.width
+    predictions[:, 0::3] = keypoints_np[:, 0::2] / x_resize * x_og
+    # put y vals back in original pixel space
+    y_resize = cfg.data.image_resize_dims.height
+    y_og = cfg.data.image_orig_dims.height
+    predictions[:, 1::3] = keypoints_np[:, 1::2] / y_resize * y_og
+    predictions[:, 2::3] = confidence_np
+
+    # get bodypart names from labeled data csv if possible
+    if ("data_dir" in cfg.data) and ("csv_file" in cfg.data):
+        csv_file = os.path.join(cfg.data.data_dir, cfg.data.csv_file)
+    else:
+        csv_file = ""
+    if os.path.exists(csv_file):
+        if "header_rows" in cfg.data:
+            header_rows = list(cfg.data.header_rows)
+        else:
+            # assume dlc format
+            header_rows = [0, 1, 2]
+        df = pd.read_csv(csv_file, header=header_rows)
+        # collect marker names from multiindex header
+        joint_labels = [c[0] for c in df.columns[1::2]]
+    else:
+        joint_labels = ["bp_%i" % n for n in range(model.num_keypoints)]
+
+    # build data frame
+    xyl_labels = ["x", "y", "likelihood"]
+    pdindex = pd.MultiIndex.from_product(
+        [["%s_tracker" % cfg.model.model_type], joint_labels, xyl_labels],
+        names=["scorer", "bodyparts", "coords"],
+    )
+    df = pd.DataFrame(predictions, columns=pdindex)
+    if save_file.endswith(".csv"):
+        df.to_csv(save_file)
+    elif save_file.find(".h") > -1:
+        df.to_hdf(save_file)
+    else:
+        raise NotImplementedError("Currently only .csv and .h5 files are supported")
+
+
+def predict_dataset(
+    cfg: DictConfig, 
+    datamod: LightningDataModule, 
+    hydra_output_directory: str,
+    ckpt_file: str,
+    heatmap_idxs: List[int] = None
+):
+    """
+    Call this function with a path to ckpt file for a trained model
+    heatmap_idxs: indexes of datapoints to save heatmaps for
+
+    """
+    model = load_model_from_checkpoint(cfg=cfg, ckpt_file=ckpt_file, eval=True)
+    model.to(_TORCH_DEVICE)
+    full_dataset = datamod.dataset
+    num_datapoints = len(full_dataset)
+    dataset_split_indices = {
+        "train": datamod.train_dataset.indices,
+        "validation": datamod.val_dataset.indices,
+        "test": datamod.test_dataset.indices,
+    }
+    with open(hydra_output_directory + "/dataset_split_indices.json", "w") as f:
+        json.dump(dataset_split_indices, f)
+
+    full_dataloader = DataLoader(dataset = full_dataset, batch_size = datamod.test_batch_size)
+    save_file = hydra_output_directory + "/predictions.csv" #default for now, should be saved to the model directory
```

Comment:
`save_file` should be made an input to the function, with default `None`. then, before line 401, add 
```
if save_file is None: 
       save_file = hydra_output_directory + "/predictions.csv"
```